web/src/components/storage/ProposalSpacePolicySection.jsx

```diff
@@ -0,0 +1,444 @@
+  return (
+    <>
+      <div>{device.name}</div>
+      <If
+        condition={device.isDrive}
+        then={<div className="fs-small">{`${device.vendor} ${device.model}`}</div>}
+      />
+    </>
+  );
+};
+
+/**
+ * Column content with information about the current content of the device.
+ * @component
+ *
+ * @param {object} props
+ * @param {StorageDevice} props.device
+ */
+const DeviceContentColumn = ({ device }) => {
+  const PartitionTableContent = () => {
+    return (
+      <div>
+        {sprintf(_("%s partition table"), device.partitionTable.type.toUpperCase())}
+      </div>
+    );
+  };
+
+  const BlockContent = () => {
+    const renderContent = () => {
+      const systems = device.systems;
+      if (systems.length > 0) return systems.join(", ");
+
+      const filesystem = device.filesystem;
+      if (filesystem?.isEFI) return _("EFI");
+      if (filesystem) return sprintf(_("%s file system"), filesystem?.type);
+
+      const component = device.component;
+      switch (component?.type) {
+        case "physical_volume":
+          return sprintf(_("LVM physical volume of %s"), component.deviceNames[0]);
+        case "md_device":
+          return sprintf(_("Member of RAID %s"), component.deviceNames[0]);
+        default:
+          return _("Not identified");
+      }
+    };
+
+    return <div>{renderContent()}</div>;
+  };
+
+  return (device.partitionTable ? <PartitionTableContent /> : <BlockContent />);
+};
+
+/**
+ * Column content with information about the size of the device.
+ * @component
+ *
+ * @param {object} props
+ * @param {StorageDevice} props.device
+ */
+const DeviceSizeColumn = ({ device }) => {
+  return <div>{deviceSize(device.size)}</div>;
+};
+
+const DeviceDetailsColumn = ({ device }) => {
+  const UnusedSize = () => {
+    if (device.filesystem) return null;
+
+    const unused = device.partitionTable?.unpartitionedSize || 0;
+
+    return (
+      <div>
+        {sprintf(_("%s unused"), deviceSize(unused))}
+      </div>
+    );
+  };
+
+  const RecoverableSize = () => {
+    const size = device.recoverableSize;
+
+    if (size === 0) return null;
+
+    return (
+      <div>
+        {sprintf(_("Shrinkable by %s"), deviceSize(device.recoverableSize))}
+      </div>
+    );
+  };
+
+  return (
+    <If condition={device.isDrive} then={<UnusedSize />} else={<RecoverableSize /> } />
+  );
+};
+
+/**
+ * Column content with the space action for a device.
+ * @component
+ *
+ * @param {object} props
+ * @param {StorageDevice} props.device
+ * @param {string} props.action - Possible values: "force_delete", "resize" or "keep".
+ * @param {boolean} [props.isDisabled=false]
+ * @param {(action: SpaceAction) => void} [props.onChange]
+ */
+const DeviceActionColumn = ({ device, action, isDisabled = false, onChange = noop }) => {
+  const changeAction = (_, action) => onChange({ device: device.name, action });
+
+  const value = (device.isDrive && action === "resize") ? "keep" : action;
```

Comment:
I think this code deserves comment why it is needed as from my first look it looks like we have some bug in underlaying code that it can return such action.